src/pages/SupplierRequisitionsPage.js

```diff
@@ -133,17 +142,17 @@ export class SupplierRequisitionsPage extends React.Component {
     }
   };
 
-  renderNewRequisitionButton = () => (
-    <PageButton
-      text={buttonStrings.new_requisition}
-      onPress={() => this.setState({ isCreatingRequisition: true })}
-    />
-  );
+  renderNewRequisitionButton = () => {
+    const { usesPrograms } = this.state;
+    const newState = usesPrograms ? { byProgramModalOpen: true } : { isCreatingRequisition: true };
+    return (
+      <PageButton text={buttonStrings.new_requisition} onPress={() => this.setState(newState)} />
+    );
+  };
 
   render() {
-    const { database, genericTablePageStyles, topRoute } = this.props;
-    const { data, isCreatingRequisition, selection } = this.state;
-
+    const { database, genericTablePageStyles, topRoute, settings } = this.props;
```

Comment:
where do settings come from ?

Reply:
redux - all of these props are

Reply:
ahh, and getAllPrograms in ByProgramModal is called with settings, now it makes sense